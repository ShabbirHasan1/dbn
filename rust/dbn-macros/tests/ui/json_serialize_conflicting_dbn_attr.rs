use dbn_macros::JsonSerialize;

#[derive(JsonSerialize)]
#[repr(C)]
struct Record {
    #[dbn(fixed_price, unix_nanos)]
    pub a: u8,
}

fn main() {}
