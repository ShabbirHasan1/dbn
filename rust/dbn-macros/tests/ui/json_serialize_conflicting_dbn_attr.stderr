error: Passed incompatible serialization arguments to dbn attr
 --> tests/ui/json_serialize_conflicting_dbn_attr.rs:6:5
  |
6 |     #[dbn(fixed_price, unix_nanos)]
  |     ^

error[E0433]: failed to resolve: unresolved import
 --> tests/ui/json_serialize_conflicting_dbn_attr.rs:3:10
  |
3 | #[derive(JsonSerialize)]
  |          ^^^^^^^^^^^^^
  |          |
  |          unresolved import
  |          help: a similar path exists: `dbn::encode`
  |
  = note: this error originates in the derive macro `JsonSerialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0603]: module `serialize` is private
 --> tests/ui/json_serialize_conflicting_dbn_attr.rs:3:10
  |
3 | #[derive(JsonSerialize)]
  |          ^^^^^^^^^^^^^
  |          |
  |          private module
  |          trait `WriteField` is not publicly re-exported
  |
note: the module `serialize` is defined here
 --> $WORKSPACE/rust/dbn/src/encode/json.rs
  |
  | pub(crate) mod serialize;
  | ^^^^^^^^^^^^^^^^^^^^^^^^
